### PR TITLE
Montana Taxable Social Security Benefits Schedule

### LINKED ISSUE #2623: Initial Montana Taxable Social Security Benefits subtraction
https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2623

### Summary
📄🧮🏔️

This pull request adds parameters and variables for calculating the Montana taxable social security benefit, which is a state-specific subtraction from the income tax base. It defines six parameters in YAML files and two variables in Python files under the `policyengine_us/parameters/gov/states/mt/tax/income/subtractions/social_security/` and `policyengine_us/variables/gov/states/mt/tax/income/subtractions/` directories, respectively. It also adds metadata, labels, units, references, and state codes for each parameter and variable.

> _`Montana` tax code_
> _Adding parameters, variables_
> _Autumn leaves of YAML_

### Walkthrough
*  Add parameters and variables for calculating Montana taxable social security benefits ([link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-b1976e3495f3a67798326d88c0f250396ec6914ee6507a3ba7a48631bca633d7L1-R21), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-f7c3d06a62a02d4cce0ee1f4a5d5949e674c0e72250998e8263feec60bcadbe7L1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-1dff81269f47c2da7c340914fb7faeaf2ccc1d22b26ba0595379e214ccd61260L1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-5559049c4f6b54ea1c2369578d50d1470e6eeb04100bd3daa371f16730ec7cfdL1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-b752b658957e74e5d09be81f0669a7b52b5f6c9df0aa1aaa5b46abdef0148168L1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-05a9766ca4c3b34aa14d072ca223005800ac431d08840d0f0d71c37157a3c16bL1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-a079a93586ee0df2541b3aa0f4c58a07f2416ddd345cc7a827541cda9a659703L1-R41), [link](https://github.com/PolicyEngine/policyengine-us/pull/2661/files?diff=unified&w=0#diff-774fb68a8f7145a655875b63c331c3742f5969507058e8cd5a62d9aea71d1b99L1-R40))

### CODE REVIEW

File: policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_modified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_modified_income.py:1]: need tests

File: policyengine_us/parameters/gov/states/mt/tax/income/subtractions/social_security/exceeding_income_cap.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+description: Montana subtracts this amount from exceeding part of modified income (compared with income cap) depending on filing status. 
+metadata:
+  label: Montana social security subtraction exceeding income cap
+  period: year
+  unit: currency-USD
+  breakdown:
+  - filing_status
+  reference:
+  - title: 2022 Montana Individual Income Tax Return Form, Taxable Social Security Benefits Schedule, Line 12
+    href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf
```
Comment (Jingj-WANG) [inline policyengine_us/parameters/gov/states/mt/tax/income/subtractions/social_security/exceeding_income_cap.yaml:10]: specify the exact page for the title and link, for example:
- title: 2022 Montana Individual Income Tax Return Form, page6, Taxable Social Security Benefits Schedule, Line 12
    href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf#page=6

File: policyengine_us/parameters/gov/states/mt/tax/income/subtractions/social_security/exceeding_income_cap.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+description: Montana subtracts this amount from exceeding part of modified income (compared with income cap) depending on filing status. 
+metadata:
+  label: Montana social security subtraction exceeding income cap
+  period: year
+  unit: currency-USD
+  breakdown:
+  - filing_status
+  reference:
```
Comment (Jingj-WANG) [inline policyengine_us/parameters/gov/states/mt/tax/income/subtractions/social_security/exceeding_income_cap.yaml:8]: add law file for Montana taxable social security benefits.

File: policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_modified_income_benefit_fraction.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        ""
+    )
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.mt.tax.income.subtractions
+        total_benefit_fraction = p.social_security.total_benefit_fraction1
+        # Line1
+        net_benefits = tax_unit.spm_unit("spm_unit_benefits", period)
+        return net_benefits * total_benefit_fraction
```
Comment (Jingj-WANG) [inline policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_modified_income_benefit_fraction.py:21]: spm_unit & tax_unit ?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_modified_income_benefit_fraction.py:21]: works as the benefits are retrieved from the spm unit

File: policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ]
+        exceeding_income_fraction = p.social_security.exceeding_income_fraction
+        extra_income_fraction = p.social_security.extra_income_fraction
+
+        net_benefits = tax_unit.spm_unit("spm_unit_benefits", period)
+        modified_income = tax_unit("mt_modified_income", period)
+
+        # if modified_income is less than cap, return 0
+        exceeding_income = modified_income - modified_income_cap
+        eligibility = exceeding_income >= 0
```
Comment (Jingj-WANG) [inline policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py:35]: where(exceeding_income >= 0, exceeding_income, 0)

File: policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (Jingj-WANG) [inline policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py:36]: exceeding_income = where(exceeding_income>=0, exceeding_income, 0)

File: policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        # Calculate the Montana Taxable Social Security Benefits
+        extra_income = max_(
+            exceeding_income - exceeding_income_cap, 0
+        )  # Line 13
+        temp1 = min_(exceeding_income, exceeding_income_cap)  # Line 14
```
Comment (Jingj-WANG) [inline policyengine_us/variables/gov/states/mt/tax/income/subtractions/mt_social_security_benefit.py:41]: rename temp1/2/3 for ease of explanation

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

**Pull request closed without merging**